### ISSUE TITLE
"+1" for useful annotations

### ISSUE THREAD

**Issue by RaoOfPhysics** [hypothesis/vision#188]
(I checked for possible duplicates, but if I missed something that discusses this issue, please close this one.)

This may tie in to moderation issues, I guess, but the ability to `+1` a particularly insightful annotation would serve a couple of different purposes:

- It would provide feedback to the author of the annotation that their contribution was useful.
- Users can track which annotations they found useful themselves by going back to their "saved" list, via the `+1`s.

The moderation angle is that a `+1` from credible people in the research field (e.g.) would add value to the annotation itself.

**Comment by fiatjaf** [comment]
Although the arguments  from @RaoOfPhysics seem reasonable. In the end this just turns hypothes.is into another social network.

**Comment by ke7ofi** [comment]
@fiatjaf That's exactly what I was thinking. At that point, we could just write a browser extension to display reddit comments.

**Comment by dwhly** [comment]
> In the end this just turns hypothes.is into another social network.

I'm curious. You've made an implicit judgment here, but not an explicit one. Clearly the tool is social already. It's not yet quite a "network" because we don't let people form direct relationships (by "friending" or "following" others)-- and by the way the +1 probably isn't the thing that makes it a network. Twitter would still be a social network, even without the (heart).

More broadly though-- what is it that you perceive Hypothes.is is and isn't, and what would you like more of and what would you like less of.

Many people ask for some form of voting or "liking".  Is there a way that this might be done that would be more to your preference?

> At that point, we could just write a browser extension to display reddit comments.

I agree that Reddit comments should be able to be seen in proximity to their target articles.  https://github.com/hypothesis/vision/issues/40

**Comment by jgmac1106** [comment]
The upvotes could also handle an issue of what happens when too many people
are annotating the same document. In my class I have some text with
hundreds of annotations (we are going to transition to groups to cover
this) and I could imagine some documents reaching that threshold.

Sometime the tool crashes under the weight of the noise. Ignore the
technical debt, that can be solved, the real issue is how would a user find
the signal through the noise. A +1 system might allow some interesting ways
curate annotations and surface better signals.

**Comment by dwhly** [comment]
> A +1 system might allow some interesting ways curate annotations and surface better signals.

As volume increases, we'll need something :)

**Comment by akuckartz** [comment]
:+1: for +1

**Comment by capitocapito** [comment]
I agree with the above points in favor of a +1 option, and the thought that hypothes.is is already a social network of sorts and thinking of it in those terms might help us come up with metaphors that make this all palatable for users (like, being able to easily "follow" or "highlight" other users).

**Comment by capitocapito** [comment]
I agree with the above points in favor of a +1 option, and the thought that hypothes.is is already a social network of sorts and thinking of it in those terms might help us come up with metaphors that make this all palatable for users (like, being able to easily "follow" or "highlight" other users).

**Comment by robertknight** [comment]
> Sometime the tool crashes under the weight of the noise.
> Ignore the technical debt, that can be solved,

@jgmac1106 - As an aside and not related to the main thread here, I've been doing some investigation into this. If you have examples of such pages that might be useful for testing.

**Comment by jgmac1106** [comment]
@robertnight do you have an issue filed already? Don't want to clog up +1
discussion.

**Comment by davepmarshall** [comment]
If you are using this as an internal tool for comments on your own web materials, it could be useful to see the most popular. High popularity might indicate the quickest wins.

**Comment by dklebedev** [comment]
@dwhly I am pondering on cons and pros of karma. Karma allows sorting by top, hot and controversial. On the other hand karma opens doors for karmawhores, abuse and is prone to selection bias. But I do not know of a better solution to discern signals by quality.

I believe Reddit does it best and I agree that there is nothing wrong with people being able to discuss anything on the internet in a similar fashion, but in context. It seems to me that a vision of hypothes.is as "reddit in context" is a great one.

Hopefully I made sense.

**Comment by klemay** [comment]
Another request for this in the Chrome webstore - tracking internally here: https://hypothesis.zendesk.com/agent/tickets/3271
